### DISCUSSION TITLE
Root package domain

### DISCUSSION THREAD

**Discussion by pisicode** [benbjohnson/wtf#3] [Q&A] [answered]
Hello Ben!

This is going to be a short question related to this blog post: https://www.gobeyond.dev/packages-as-layers/

What are your thoughts on potentially moving the domain files from the root directory into their own package (let's call it 'domain' for the sake of simplicity)? So that the root level of the project contains only a readme, build/deploy scripts etc. and not the actual code (as I believe a much larger application would use many .go files to declare its domain).

**Comment by benbjohnson** [comment, marked as answer]
@bgospodaru That's certainly an option to move to another package although I don't especially like the name `domain`. That being said, I don't personally mind a lot of Go files in the root for a large project.

**Reply by menzhessarov** [reply]
Why don't you like the name `domain`?

**Reply by benbjohnson** [reply]
I don't like the name `domain` for two reasons:

1. It doesn't read very well. Calling something `domain.Customer` is awkward whereas `myapp.Customer` describes it as a "customer of myapp".

2. If your application gets larger you may want to split out into subdomains but also keep a shared top-level domain. In that case it's problematic since you'd end up with multiple `domain` packages that could be imported by the same file.
